### PR TITLE
Migrate to `pyproject.toml`

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/11164?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11164/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11164
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Remove `setup.py` and migrate to `pyproject.toml`.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .MANIFEST.in
Comment (harupy) [inline .MANIFEST.in:1]: Renamed to ensure build succeeds without this file and setup.py. Will be removed.

File: pyproject.toml
Comment (harupy) [inline pyproject.toml:48]: This needs to be fixed.

File: pyproject.toml
Comment (harupy) [inline pyproject.toml:1]: As a follow-up, we need to add a check to ensure both `pyproject.toml` and `pyproject.skinny.toml` is updated.

File: .setup.py
Comment (harupy) [inline .setup.py:1]: will be removed, but wanted to keep it during development.

File: dev/pyproject.py
Comment (serena-ruan) [inline dev/pyproject.py:138]: Is this `mlflow/models/container/*`?

File: dev/pyproject.py
Comment (serena-ruan) [inline dev/pyproject.py:136]: Miss `mlflow/` prefix? Or does it work as well?

**Review by serena-ruan — APPROVED** [review]
LGTM!

File: dev/pyproject.py
Comment (harupy) [reply dev/pyproject.py:136]: `mlflow/` must be stripped.

File: dev/pyproject.py
Comment (BenWilson2) [inline dev/pyproject.py:19]: nice idea for catching indented comment lines! +1

File: dev/pyproject.py
Comment (BenWilson2) [inline dev/pyproject.py:39]: Can we change this to the maintainers group email address?

File: dev/pyproject.py
Comment (BenWilson2) [inline dev/pyproject.py:48]: Should we take this opportunity to fix our classifiers finally?

```python
"classifiers": [
    "Development Status :: 5 - Production/Stable",
    "Intended Audience :: Developers",
    "Intended Audience :: End Users/Desktop",
    "Intended Audience :: Science/Research",
    "Intended Audience :: Information Technology",
    "Topic :: Scientific/Engineering :: Artificial Intelligence",
    "Topic :: Software Development :: Libraries :: Python Modules",
    "License :: OSI Approved :: Apache Software License",
    "Operating System :: OS Independent",
    "Programming Language :: Python :: {python_version}",
]
```

File: pyproject.toml
Comment (BenWilson2) [reply pyproject.toml:48]: Let's update to maintainers email address

**Review by BenWilson2 — APPROVED** [review]
LGTM once remaining comments are addressed! Great work Haru!

File: dev/pyproject.py
Comment (harupy) [reply dev/pyproject.py:39]: Sounds good. I'll use maintainers instead of authors.

File: pyproject.toml
Comment (harupy) [reply pyproject.toml:48]: sounds good!

File: dev/pyproject.py
Comment (harupy) [reply dev/pyproject.py:148]: stripped!

**Pull request merged**
